rootston/desktop.c

```diff
@@ -442,9 +452,33 @@ void view_init(struct roots_view *view, struct roots_desktop *desktop) {
 	wl_signal_add(&view->wlr_surface->events.new_subsurface,
 		&view->new_subsurface);
 
+	wl_list_insert(&view->desktop->views, &view->link);
 	view_damage_whole(view);
 }
 
+void view_unmap(struct roots_view *view) {
+	assert(view->wlr_surface != NULL);
+
```

Comment:
Shouldn't there be a `wlr_signal_emit_safe(&view->events.unmap, view);` here?
Quick text-search on the diff page doesn't bring up a signal_emit for unmap on the view, only for surfaces.

Reply:
You're right!